--- a/drivers/xen/manage.c	2026-03-24 21:32:50.767291118 +0100
+++ b/drivers/xen/manage.c	2026-03-24 21:32:50.769291117 +0100
@@ -7,13 +7,16 @@
 	xen_arch_post_suspend(si->cancelled);
 	xen_manage_runstate_time(si->cancelled ? 1 : 0);
 	gnttab_resume();
+	{ extern void isol8_trace(u64 bit); isol8_trace(1ULL << 9); }
 
 	if (!si->cancelled) {
 		xen_irq_resume();
 		xen_timer_resume();
 	}
+	{ extern void isol8_trace(u64 bit); isol8_trace(1ULL << 10); }
 
 	syscore_resume();
+	{ extern void isol8_trace(u64 bit); isol8_trace(1ULL << 11); }
 
 	return 0;
 }
