--- a/drivers/xen/manage.c	2026-03-26 12:12:06.684612382 +0000
+++ b/drivers/xen/manage.c	2026-03-26 12:31:21.995011249 +0000
@@ -75,6 +75,12 @@
                                            ? virt_to_gfn(xen_start_info)
                                            : 0);
 
+	/* isol8: volatile write to stack local */
+	{
+		volatile unsigned long dummy = 0xDEAD;
+		(void)dummy;
+	}
+
 	xen_arch_post_suspend(si->cancelled);
 	xen_manage_runstate_time(si->cancelled ? 1 : 0);
 	gnttab_resume();
