--- a/kernel/stop_machine.c
+++ b/kernel/stop_machine.c
@@ -225,6 +225,10 @@
 	do {
 		/* Chill out and ensure we re-read multi_stop_state. */
 		stop_machine_yield(cpumask);
+		{
+			extern void isol8_trace(u64 bit);
+			isol8_trace(1ULL << 12);
+		}
 		newstate = READ_ONCE(msdata->state);
 		if (newstate != curstate) {
 			curstate = newstate;
@@ -236,6 +240,10 @@
 			case MULTI_STOP_RUN:
 				if (is_active)
 					err = msdata->fn(msdata->data);
+				{
+					extern void isol8_trace(u64 bit);
+					isol8_trace(1ULL << 13);
+				}
 				break;
 			default:
 				break;
