--- a/kernel/stop_machine.c	2026-03-25 21:54:34.739888704 +0000
+++ b/kernel/stop_machine.c	2026-03-26 20:04:04.134453095 +0000
@@ -198,6 +198,11 @@
 }
 
 /* This is the cpu_stop function which stops the CPU. */
+static noinline void pv_resume_fence(void)
+{
+	asm volatile("" : : : "memory");
+}
+
 static int multi_cpu_stop(void *data)
 {
 	struct multi_stop_data *msdata = data;
@@ -241,6 +246,7 @@
 				break;
 			}
 			ack_state(msdata);
+			pv_resume_fence();
 		} else if (curstate > MULTI_STOP_PREPARE) {
 			/*
 			 * At this stage all other CPUs we depend on must spin
