--- a/kernel/stop_machine.c	2026-03-25 21:54:34.739888704 +0000
+++ b/kernel/stop_machine.c	2026-03-26 17:41:27.296960293 +0000
@@ -225,6 +225,10 @@
 	do {
 		/* Chill out and ensure we re-read multi_stop_state. */
 		stop_machine_yield(cpumask);
+		{
+			extern void isol8_trace(u64 bit);
+			isol8_trace(1ULL << 12);
+		}
 		newstate = READ_ONCE(msdata->state);
 		if (newstate != curstate) {
 			curstate = newstate;
@@ -236,11 +240,19 @@
 			case MULTI_STOP_RUN:
 				if (is_active)
 					err = msdata->fn(msdata->data);
+				{
+					extern void isol8_trace(u64 bit);
+					isol8_trace(1ULL << 13);
+				}
 				break;
 			default:
 				break;
 			}
 			ack_state(msdata);
+			{
+				extern void isol8_trace(u64 bit);
+				isol8_trace(1ULL << 27);
+			}
 		} else if (curstate > MULTI_STOP_PREPARE) {
 			/*
 			 * At this stage all other CPUs we depend on must spin
@@ -251,9 +263,18 @@
 			/* Also suppress RCU CPU stall warnings. */
 			rcu_momentary_eqs();
 		}
+		{
+			extern void isol8_trace(u64 bit);
+			if (curstate == MULTI_STOP_EXIT)
+				isol8_trace(1ULL << 29);
+		}
 	} while (curstate != MULTI_STOP_EXIT);
 
 	local_irq_restore(flags);
+	{
+		extern void isol8_trace(u64 bit);
+		isol8_trace(1ULL << 30);
+	}
 	return err;
 }
 
